server/helpers.go

```diff
@@ -232,3 +232,95 @@ func authWrap(fn http.Handler, username, password string) http.HandlerFunc {
 		fn.ServeHTTP(w, r)
 	}
 }
+
+func getTaskInfoV0(t eremetic.Task, conf *config.Config, id string, w http.ResponseWriter, r *http.Request) {
+	if strings.Contains(r.Header.Get("Accept"), "text/html") {
+		renderHTML(w, r, t, id, conf)
+	} else {
+		task := eremetic.TaskV0FromTask(&t)
+		if reflect.DeepEqual(task, (eremetic.TaskV0{})) {
+			writeJSON(http.StatusNotFound, nil, w)
+			return
+		}
+		writeJSON(http.StatusOK, task, w)
+	}
+}
+
+func getTaskInfoV1(t eremetic.Task, conf *config.Config, id string, w http.ResponseWriter, r *http.Request) {
+	task := eremetic.TaskV1FromTask(&t)
+	if reflect.DeepEqual(task, (eremetic.TaskV1{})) {
+		writeJSON(http.StatusNotFound, nil, w)
+		return
+	}
+	writeJSON(http.StatusOK, task, w)
+}
+
+func addTaskV0(conf *config.Config, scheduler eremetic.Scheduler, w http.ResponseWriter, r *http.Request) {
+	var request0 eremetic.RequestV0
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 1048576))
+	if err != nil {
+		handleError(err, w, "Unable to read payload.")
+		return
+	}
+	err = json.Unmarshal(body, &request0)
+	if err != nil {
+		handleError(err, w, "Unable to parse body into a valid request.")
+		return
+	}
+
+	request := eremetic.RequestFromV0(request0)
+	taskID, err := scheduler.ScheduleTask(request)
+	if err != nil {
+		logrus.WithError(err).Error("Unable to create task.")
+		httpStatus := 500
+		if err == eremetic.ErrQueueFull {
+			httpStatus = 503
+		}
+		errorMessage := errorDocument{
+			err.Error(),
+			"Unable to schedule task",
+		}
+		writeJSON(httpStatus, errorMessage, w)
+		return
+	}
+
+	w.Header().Set("Location", absURL(r, fmt.Sprintf("/task/%s", taskID), conf))
+	writeJSON(http.StatusAccepted, taskID, w)
+}
+
+func addTaskV1(conf *config.Config, scheduler eremetic.Scheduler, w http.ResponseWriter, r *http.Request) {
+	var request0 eremetic.RequestV1
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 1048576))
+	if err != nil {
+		handleError(err, w, "Unable to read payload.")
+		return
+	}
+	err = json.Unmarshal(body, &request0)
```

Comment:
It's quite a bit of duplication when the type used here is pretty much the only difference. Could the rest be extracted to a shared hepler?